pkg/api/alert_client.go

```diff
@@ -23,6 +71,97 @@ type AlertSettingsConditionPayload struct {
 	Channels    []string `json:"channels" yaml:"channels"`
 }
 
+func constructQueryString(page int, pageSize int, sortField, sortOrder, templateName, channelName string) (query string) {
+	path := fmt.Sprintf("%s?pageSize=%d", pkg.AlertChannelsPath, pageSize)
+	if page != 0 {
+		path = path + fmt.Sprintf("&page=%d", page)
+	}
+	if sortField != "" {
+		path = path + fmt.Sprintf("&sortField=%s", sortField)
+	}
+	if sortOrder != "" {
+		path = path + fmt.Sprintf("&sortOrder=%s", sortOrder)
+	}
+	if templateName != "" {
+		path = path + fmt.Sprintf("&templateName=%s", templateName)
+	}
+	if channelName != "" {
+		path = path + fmt.Sprintf("&channelName=%s", channelName)
+	}
+	return path
+}
+
+// GetAlertChannels handles the API call get the list of alert channels
+func (c *Client) GetAlertChannels(page int, pageSize int, sortField, sortOrder, templateName, channelName string) (response AlertChannelResponse, err error) {
+	path := constructQueryString(page, pageSize, sortField, sortOrder, templateName, channelName)
+	resp, err := c.Do(http.MethodGet, path, contentTypeJSON, nil)
+	if err != nil {
+		return
+	}
+	err = c.ReadJSON(resp, &response)
+	return
+}
+
+// GetAlertChannelsWithDetails handles the API call get the list of alert channels with details
```

Comment:
Again, is this needed?

Reply:
As discussed this is needed due to the different response signature